source/Halibut/Transport/Protocol/TemporaryFileStream.cs

```diff
@@ -39,15 +39,22 @@ public void SaveTo(string filePath)
 
         void SetFilePermissionsToInheritFromParent(string filePath)
         {
-            var fileInfo = new FileInfo(filePath);
+            try
+            {
+                var fileInfo = new FileInfo(filePath);
 #pragma warning disable PC001 // API not supported on all platforms
-            var fileSecurity = fileInfo.GetAccessControl();
+                var fileSecurity = fileInfo.GetAccessControl();
 
-            //When isProtected (first param) is false, SetAccessRuleProtection changes the permissions of the file to allow inherited permissions. 
-            //preserveInheritance (second param) is ignored when isProtected is false.
-            fileSecurity.SetAccessRuleProtection(false, false);
-            fileInfo.SetAccessControl(fileSecurity);
+                //When isProtected (first param) is false, SetAccessRuleProtection changes the permissions of the file to allow inherited permissions. 
+                //preserveInheritance (second param) is ignored when isProtected is false.
+                fileSecurity.SetAccessRuleProtection(false, false);
+                fileInfo.SetAccessControl(fileSecurity);
 #pragma warning restore PC001 // API not supported on all platforms
+            }
+            catch (UnauthorizedAccessException)
+            {
+                // ignored, user has likely disallowed full control over their artifact directory 
```

Comment:
Feels like we should be logging this comment here, even if it's just a verbose log, so customers have some hint that it's falling into this code path. I guess the flip side of this, is there a good reason not to log it?

Reply:
:thinking: Good point, as a diagnostic patch we originally did a bunch of logging (to `OctopusServer.txt`) to ensure we had the right fix. I ripped it out after a chat with some other team members as we felt that that 'cut' was probably too noisy. 

That said, I think you've identified a good middle ground, in the event something unexpected occurs, it would be nice to be able to capture this intent in the logs somewhere. 

Thanks! :bowing_man: